catalog/resource_metastore.go

```diff
@@ -89,32 +59,120 @@ func ResourceMetastore() *schema.Resource {
 			}
 			return m
 		})
-	update := updateFunctionFactory("/unity-catalog/metastores", []string{"owner", "name", "delta_sharing_scope",
-		"delta_sharing_recipient_token_lifetime_in_seconds", "delta_sharing_organization_name"})
 
 	return common.Resource{
 		Schema: s,
 		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
-			var create CreateMetastore
+			var create catalog.CreateMetastore
+			var update catalog.UpdateMetastore
 			common.DataToStructPointer(d, s, &create)
-			mi, err := NewMetastoresAPI(ctx, c).createMetastore(create)
-			if err != nil {
-				return err
+			common.DataToStructPointer(d, s, &update)
+			if c.Config.IsAccountClient() && c.Config.AccountID != "" {
```

Comment:
Also, I wonder if it would be useful to define a function
```
func (c *client.DatabricksClient) WorkspaceOrAccountRequest[T any](wsCallback func (*databricks.WorkspaceClient) (T, error), accCallback func (*databricks.AccountClient) (T, error)) (T, error) {
  if c.Config.IsAccountClient() {
    a, err := c.AccountClient()
    if err != nil {
      return nil, err
    }
    return accCallback(a)
  } else {
    ws, err := c.WorkspaceClient()
    if err != nil {
      return nil, err
    }
    return wsCallback(ws)
  }
}
```
This would save a lot of boilerplate.

Reply:
this needs to go into Go SDK as `DatabricksClient` is defined there. Also Go would not support specifying type parameters on methods 😢

Reply:
You learn something new every day :)

I was talking about the client in common/client.go in this repository. I suppose this would have to be a plain function to do this or to simply use `any` as the return type of the function and do a type assertion. What do you think?